src/libstd/panicking.rs

```diff
@@ -412,10 +411,10 @@ pub fn begin_panic<M: Any + Send>(msg: M, file_line_col: &(&'static str, u32, u3
     }
 
     unsafe impl<A: Send + 'static> BoxMeUp for PanicPayload<A> {
-        fn box_me_up(&mut self) -> *mut (dyn Any + Send) {
+        fn take_box(&mut self) -> *mut (dyn Any + Send) {
             let data = match self.inner.take() {
                 Some(a) => Box::new(a) as Box<dyn Any + Send>,
-                None => Box::new(()),
+                None => Box::new(()), // this should never happen: we got called twice
```

Comment:
I was wondering if it is appropriate to call `process::abort()` here?

Reply:
Maybe. I feel it’s not very important since this is a private API that has exactly one call site.

Reply:
Let me re-phrase my question: is there any reason *not* to do that? Is there any legitimate use of this interface that calls `take_box` twice?

Reply:
Actually what I am wondering about even more is what the method does not take `self` by-value?

Reply:
This is an impl for a private type `PanicPayload`, whose value is only ever constructed in one place. A pointer to that value is `__rust_start_panic`, which we can consider a private implementation detail of the standard library. So both the definition and all uses of this API (all one of them) are under our control. So it doesn’t matter in my opinion what the code does in situations that we  know never happen. There is no reason to abort, and there is no reason not to abort. Feel free to change this to abort if you’d like, I don’t mind r+’ing that change.

This method cannot take `self` by value because it is (only, ever) called by `src/libpanic_unwind/lib.rs`’s `__rust_start_panic` which dereferences a raw pointer (cast from `usize`) to get `&mut &mut dyn BoxMeUp`. It cannot (easily) be an owned `Box<dyn BoxMeUp>` because libpanic_unwind doesn’t depend on liballoc which defines `Box`. Oh and for the trait to be object-safe it might have to be `self: Box<Self>` or something.

Reply:
>  It cannot (easily) be an owned Box<dyn BoxMeUp> because libpanic_unwind doesn’t depend on liballoc which defines Box

Ah, thanks, I added a comment along those lines (and made it abort). Please check.